packages/langium/src/grammar/type-system/inferred-types.ts

```diff
@@ -312,18 +307,11 @@ function addAssignment(current: TypePart, assignment: Assignment): void {
     const typeItems: TypeCollection = { types: new Set(), reference: false };
     findTypes(assignment.terminal, typeItems);
 
-    let typeAlternatives: PropertyType[] = [];
-    if (assignment.operator === '=' && !typeItems.reference) {
```

Comment:
What happened with this case here? Is it obsolete?

Reply:
This case has to be covered by new code.

Reply:
I'm also a bit curious why this was dropped. You added this originally as part of #455 , but I'm not exactly sure what this part was added for. Given this an assignment, and this is the `addAssignment` function, I would think we would want to retain this. Is there a reason why not?

Reply:
Previous logic was:
* if we have an array or/and a reference of type alternatives, we pack them in one list
* if we have only type alternatives (not under an array or a reference), we consider each type separately and get many 1-element-lists

Now I've added the `toPropertyType` function, that implements this logic, so we don't need to keep these code anymore.

Reply:
Ahh, got it, I see you've lifted it up into the common function. Nice 😎